pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -905,8 +910,23 @@ func (vs *VSphere) CreateVolume(name string, size int, tags *map[string]string)
 	dc, err := f.Datacenter(ctx, vs.cfg.Global.Datacenter)
 	f.SetDatacenter(dc)
 
+	if (*tags)["adapterType"] == "" {
+		(*tags)["adapterType"] = DefaultAdapterType
+	}
+	if (*tags)["diskType"] == "" {
+		(*tags)["diskType"] = DefaultDiskType
+	}
+
+	// vmdks will be created inside kubevols dir
+	kubevolsdir := "[" + vs.cfg.Global.Datastore + "] " + VolDir + "/"
```

Comment:
use Govmomi finder object to build path within a datastore, for example:
https://github.com/hashicorp/terraform/blob/20ed831060097e7cedef4d0fcb78440e2579cb2e/builtin/providers/vsphere/resource_vsphere_virtual_disk.go#L126-L131